macOS/DuckDuckGo/DataImport/Bookmarks/Firefox/FirefoxHistoryReader.swift

```diff
@@ -95,11 +98,18 @@ final class FirefoxHistoryReader {
             try FirefoxFrecentSite.fetchAll(database, sql: allFrecentSitesQuery())
         }
 
-        /// Remove invalid URLs
-        let validFrecentSites = frecentSites.filter { site in
-            guard let url = URL(string: site.url), let host = url.host else { return false }
-            return (url.isHttps || url.isHttp) && !searchHosts.contains(where: { host.contains($0) })
-        }
+        let validFrecentSites = frecentSites
+            .filter { site in
+                // Remove invalid URLs and match Firefox filtering (only HTTP/HTTPS URLs and not search engine hosts).
+                guard let url = URL(string: site.url), let host = url.host else { return false }
+                return (url.isHttps || url.isHttp) && !searchHosts.contains(where: { host.droppingWwwPrefix().hasPrefix("\($0).") })
+            }.map { site in
+                // If the site URL is from a site with search, replace it with the corresponding shortcut URL.
+                if let shortcutURL = searchShortcuts.first(where: { site.url.contains($0.key) })?.value {
```

Comment:
is it correct? can't it mistakenly replace the value for a url like "googlefly.com"?

Reply:
Argh, I fixed that locally and then lost the fix while moving some things around. I’ll fix that and add a test to make sure the problem isn’t reintroduced.